### PR TITLE
Add support for Rosstech Digital Control Unit DCU-706/Sundance/Jacuzzi

### PR DESCRIPTION
**Pull request by suaveolent**
This PR adds support for the Rosstech DU-706.

The DCU can be found in many Sundance / Jacuzzi Spas. It periodically (11s) transmits the temperature of the spa.

Sample data can be found here: https://github.com/merbanan/rtl_433_tests/pull/442

### CODE REVIEW

File: src/devices/rosstech_dcu706.c [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+
+#include "decoder.h"
+
+/**
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:14]: If this doc comment is not immediately before the function then use
`/** @fn int rosstech_dcu706_decode(r_device *decoder, bitbuffer_t *bitbuffer)`

File: src/devices/rosstech_dcu706.c [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        }
+
+        if (count % 2 == 0) {
+            checksum |= (1 << bit);
+        }
+    }
+
+    return checksum;
+}
+
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:56]: remove multiple newlines

File: src/devices/rosstech_dcu706.c [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    uint8_t msg[7];
+
+    if (bitbuffer->num_rows != 1
+            || bitbuffer->bits_per_row[0] < 55
+            || bitbuffer->bits_per_row[0] > 300) {
+        decoder_logf(decoder, 2, __func__, "bit_per_row %u out of range", bitbuffer->bits_per_row[0]);
+        return DECODE_ABORT_EARLY; // Unrecognized data
+    }
+
+    uint8_t const preambleDataTransmission[] = {0xDD, 0x40};
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:72]: this should be the first line of the function

File: src/devices/rosstech_dcu706.c [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+    unsigned start_pos = bitbuffer_search(bitbuffer, 0, 0, preambleDataTransmission, 11);
+
+    if (start_pos == bitbuffer->bits_per_row[0]) {
+
+        // The Bond command also contains the temperature
+        uint8_t const preambleBond[] = {0xCD, 0x00};
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:79]: should be top of the function too

File: src/devices/rosstech_dcu706.c [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        return DECODE_ABORT_LENGTH;
+    }
+
+    bitbuffer_extract_bytes(bitbuffer, 0, start_pos, msg, sizeof(msg) * 8);
+
+    uint8_t syncType = (msg[0] << 1) | (msg[1] >> 7); //S
+    uint8_t id_high = (msg[1] << 4 | msg[2] >> 4);
+    uint8_t id_low = (msg[2] << 7 | msg[3] >> 1);
+    uint16_t id = (uint16_t)(id_high << 8) | id_low; // I
+    uint8_t temp = msg[4] << 2 | msg[5] >> 6; // T
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:98]: use parens ( )

File: src/devices/rosstech_dcu706.c [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    uint8_t checkSum = msg[5] << 5 | msg[6] >> 3; // C
+
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:100]: one newline only

File: src/devices/rosstech_dcu706.c [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    if (calculatedChecksum != checkSum) {
+        decoder_logf(decoder, 2, __func__, "Sanity Check failed. Expected: %04x, Calculated: %04x. Maybe sanity function calculated wrong!", checkSum, calculatedChecksum);
+        // return DECODE_FAIL_SANITY;
+    }
+
+    uint8_t temp_c = ((temp-32)*5)/9;
+
+    /* clang-format off */
+    data_t *data = data_make(
+        "model",            "Model",              DATA_STRING,   "Rosstech Digital Control Unit DCU-706/Sundance/Jacuzzi",
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:119]: model needs to be e.g. `Rosstech-Spa`

File: src/devices/rosstech_dcu706.c [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+        "syncType",         "Transmission Type",  DATA_STRING,   syncType == 0xba ? transmissionTypeData : transmissionTypeBond,    
+        "id",               "ID",                 DATA_FORMAT,   "%04x",   DATA_INT,    id,
+        "temperature_C",    "Temperature",        DATA_FORMAT,   "%d °C",  DATA_INT,     temp_c,
+        "checkSum",         "Integrity",          DATA_STRING,   "Check Sum",
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:123]: use `"mic"` not `checkSum`, also value is `"CHECKSUM"`

File: src/devices/rosstech_dcu706.c [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+        // return DECODE_FAIL_SANITY;
+    }
+
+    uint8_t temp_c = ((temp-32)*5)/9;
+
+    /* clang-format off */
+    data_t *data = data_make(
+        "model",            "Model",              DATA_STRING,   "Rosstech Digital Control Unit DCU-706/Sundance/Jacuzzi",
+        "syncType",         "Transmission Type",  DATA_STRING,   syncType == 0xba ? transmissionTypeData : transmissionTypeBond,    
+        "id",               "ID",                 DATA_FORMAT,   "%04x",   DATA_INT,    id,
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:121]: move below "model" line

File: src/devices/rosstech_dcu706.c [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+        decoder_logf(decoder, 2, __func__, "Sanity Check failed. Expected: %04x, Calculated: %04x. Maybe sanity function calculated wrong!", checkSum, calculatedChecksum);
+        // return DECODE_FAIL_SANITY;
+    }
+
+    uint8_t temp_c = ((temp-32)*5)/9;
+
+    /* clang-format off */
+    data_t *data = data_make(
+        "model",            "Model",              DATA_STRING,   "Rosstech Digital Control Unit DCU-706/Sundance/Jacuzzi",
+        "syncType",         "Transmission Type",  DATA_STRING,   syncType == 0xba ? transmissionTypeData : transmissionTypeBond,
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:120]: should be just `"sync"` with an int of 0/1

File: src/devices/rosstech_dcu706.c [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+
+    decoder_output_data(decoder, data);
+    return 1;
+
+}
+
+static char const *const output_fields[] = {
+        "model",
+        "id",
+        "temp",
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:135]: `temperature_C`, also add `sync`

File: src/devices/rosstech_dcu706.c [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+        "mic",
+        NULL,
+};
+
+r_device const rosstech_dcu706 = {
+        .name        = "Rosstech Digital Control Unit DCU-706/Sundance/Jacuzzi",
+        .modulation  = OOK_PULSE_PCM,
+        .short_width = 200,
+        .long_width  = 200,
+        .sync_width  = 0, // 1:10, tuned to widely match 2450 to 2850
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:145]: remove this line

File: src/devices/rosstech_dcu706.c [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+        .reset_limit = 2000,
+        .decode_fn   = &rosstech_dcu706_decode,
+        .fields      = output_fields,
+};
```
Comment (zuckschwerdt) [inline src/devices/rosstech_dcu706.c:149]: add EOF newline

File: src/devices/rosstech_dcu706.c (hunk 10 above)
Comment (suaveolent) [reply src/devices/rosstech_dcu706.c:120]: Removed sync alltogether, since it is always present.

**Comment by zuckschwerdt** [comment]
Looking good. I'll update the PR slightly then merge. Thanks.
(e.g. we don't want `uint8_t temp_c = ((temp-32)*5)/9;` but `temperature_F` and the `-C si` option)

**Comment by suaveolent** [comment]
Great. Thank you! 🎊

**Pull request merged**